CommonTools/RecoAlgos/interface/CosmicTrackingParticleSelector.h

```diff
@@ -103,13 +106,10 @@ class CosmicTrackingParticleSelector {
     //if (tpr->pdgId()==pdgId_[it]) testId = true;
     //}
 
-    edm::ESHandle<TrackerGeometry> tracker;
-    iSetup.get<TrackerDigiGeometryRecord>().get(tracker);
-    edm::ESHandle<GlobalTrackingGeometry> theGeometry;
-    iSetup.get<GlobalTrackingGeometryRecord>().get(theGeometry);
+    edm::ESHandle<TrackerGeometry> tracker = iSetup.getHandle(trackerDigiGeomToken_);
+    edm::ESHandle<GlobalTrackingGeometry> theGeometry = iSetup.getHandle(globalTrackingGeomToken_);
 
-    edm::ESHandle<MagneticField> theMF;
-    iSetup.get<IdealMagneticFieldRecord>().get(theMF);
+    edm::ESHandle<MagneticField> theMF = iSetup.getHandle(theMFToken_);
```

Comment:
Since the validity of the `ESHandle` is not checked perhaps one can retrieve directly the product.